src/transformers/models/paligemma/modeling_paligemma.py

```diff
@@ -484,7 +487,7 @@ def forward(
                 # we use the input attention mask to shift the logits and labels, because it is 2D.
                 shift_attention_mask = input_attention_mask[..., 1:]
                 shift_logits = shift_logits[shift_attention_mask.to(logits.device) != 0].contiguous()
-                shift_labels = shift_labels[shift_attention_mask.to(logits.device) != 0].contiguous()
+                shift_labels = shift_labels[shift_attention_mask.to(shift_labels.device) != 0].contiguous()
```

Comment:
this could make sense, but it's weird that all the other models don't need this. Not sure how the device of the labels is chosen in accelerate.

Reply:
I think it's done like this in `Llava` as well for instance:
```python
shift_logits = logits[..., :-1, :][shift_attention_mask.to(logits.device) != 0].contiguous()
shift_labels = labels[..., 1:][shift_attention_mask.to(labels.device) != 0].contiguous()
```

Reply:
ok!